### PR TITLE
CloudWatch: ContainerInsights metrics support

### LINKED ISSUE #18970: ContainerInsights support for Cloudwatch Logs
**What would you like to be added**:
Submitted PR for this feature: 
https://github.com/grafana/grafana/pull/18971

**Why is this needed**:
This is needed for https://docs.amazonaws.cn/en_us/AmazonCloudWatch/latest/monitoring/Container-Insights-metrics-EKS.html
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html

### PR DESCRIPTION
**Pull request by fpapalardo**
**What this PR does / why we need it**:
This is needed for https://docs.amazonaws.cn/en_us/AmazonCloudWatch/latest/monitoring/Container-Insights-metrics-EKS.html
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html

**Which issue(s) this PR fixes**:

Fixes #18970 

**Special notes for your reviewer**:
Screen Shot 2019-09-09 at 9 49 21 AM
Screen Shot 2019-09-09 at 9 50 10 AM

### CODE REVIEW

File: pkg/tsdb/cloudwatch/metric_find_query.go [hunk 1]
```diff
@@ -196,6 +197,7 @@ func init() {
 		"AWS/TrustedAdvisor":    {},
 		"AWS/VPN":               {"TunnelIpAddress", "VpnId"},
 		"AWS/WorkSpaces":        {"DirectoryId", "WorkspaceId"},
+		"ContainerInsights":     {"ClusterName", "Namespace", "InstanceId", "PodName", "Service"},
```
Comment (papagian) [inline pkg/tsdb/cloudwatch/metric_find_query.go]: I think that `NodeName` should be added too.

**Review by papagian — CHANGES_REQUESTED** [review]
Thank you for contributing!
Please, check my comment whether `NodeName` should be included in the Dimensions.

File: pkg/tsdb/cloudwatch/metric_find_query.go (hunk 1 above)
Comment (francopeapea) [reply pkg/tsdb/cloudwatch/metric_find_query.go]: Resolved, in next commit

**Comment by papagian** [comment]
Could you please sync your branch with the grafana master in order the CI scripts will run successfully? thank you

**Comment by francopeapea** [comment]
Merged master with my branch to be up to date

**Review by papagian — APPROVED** [review]
Thank you for contributing!

**Comment by nitrag** [comment]
Newb here, I'm running 6.4.1 and I cannot see ContainerInsights in the list of metrics. Very excited for this!

image

**Comment by francopeapea** [comment]
This seems to be a milestone for 6.5, but it’s going to slow down a lot of people that want to get cloud watch container insights

**Comment by marefr** [comment]
@francopeapea think you should be able to input the metric/dimensions manually until then or use a nightly build of Grafana.

**Comment by nitrag** [comment]
I installed the nightly, thanks for that suggestion. I do not get any data however if I use the pod dimension, or perhaps it's the combination of cluster

```json
{
  "xhrStatus": "complete",
  "request": {
    "method": "POST",
    "url": "api/tsdb/query",
    "data": {
      "from": "1570374100720",
      "to": "1570460500720",
      "queries": [
        {
          "refId": "A",
          "intervalMs": 120000,
          "maxDataPoints": 869,
          "datasourceId": 1,
          "type": "timeSeriesQuery",
          "namespace": "ContainerInsights",
          "metricName": "pod_memory_utilization",
          "statistics": [
            "Average"
          ],
          "dimensions": {
            "ClusterName": "pv-platform-master",
            "PodName": "core-api"
          },
          "period": "60",
          "region": "us-east-1",
          "id": "",
          "expression": "",
          "highResolution": false,
          "datasource": "CloudWatch"
        }
      ]
    }
  },
  "response": {
    "results": {
      "A": {
        "refId": "A",
        "meta": {},
        "series": [
          {
            "name": "pod_memory_utilization_Average",
            "points": [],
            "tags": {
              "ClusterName": "pv-platform-master",
              "PodName": "core-api"
            }
          }
        ],
        "tables": null
      }
    }
  }
}
```

or

```json
{
  "xhrStatus": "complete",
  "request": {
    "method": "POST",
    "url": "api/tsdb/query",
    "data": {
      "from": "1570374262738",
      "to": "1570460662738",
      "queries": [
        {
          "refId": "A",
          "intervalMs": 120000,
          "maxDataPoints": 869,
          "datasourceId": 1,
          "type": "timeSeriesQuery",
          "namespace": "ContainerInsights",
          "metricName": "pod_memory_utilization",
          "statistics": [
            "Average"
          ],
          "dimensions": {
            "PodName": "core-api"
          },
          "period": "60",
          "region": "us-east-1",
          "id": "",
          "expression": "",
          "highResolution": false,
          "datasource": "CloudWatch"
        }
      ]
    }
  },
  "response": {
    "results": {
      "A": {
        "refId": "A",
        "meta": {},
        "series": [
          {
            "name": "pod_memory_utilization_Average",
            "points": [],
            "tags": {
              "PodName": "core-api"
            }
          }
        ],
        "tables": null
      }
    }
  }
}
```

**Comment by marefr** [comment]
@nitrag please verify you can see the metrics in AWS CloudWatch console using same time range and dimensions. Also try to expand dashboard time range in Grafana just in case you got that wrong,

**Comment by nitrag** [comment]
Very interesting. CloudWatch, or rather the query against Container Insights data in CloudWatch, requires that I also specify `Namespace`. (This is EKS data). A rather silly requirement on AWS side that it doesn't aggregate. I added the namespace dimension and it began working. So in total, `ClusterName` and `Namespace` is required in addition to `PodName`.

image


My apologies for not debugging in CloudWatch enough initially to cause this stir. Thanks so much for adding this feature!!!

**Pull request merged**